pages/api/analytics/species-planted.ts

```diff
@@ -0,0 +1,38 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import db from '../../../src/utils/connectDB';
+
+export default async function handler(
+  req: NextApiRequest,
+  response: NextApiResponse
+) {
+  const { projectId, startDate, endDate } = req.body;
+
+  if (req.method === 'POST') {
+    try {
+      const res = await db.query(
+        'SELECT \
+            pl.plant_project_id, \
+            pl.plant_date, \
+            ps.tree_count, \
+            ps.scientific_species_id, \
+            ss.name, \
+            ps.other_species \
+            FROM planted_species ps \
+        INNER JOIN plant_location pl ON ps.plant_location_id = pl.id \
+        LEFT JOIN scientific_species ss ON ps.scientific_species_id = ss.id \
+        WHERE pl.plant_project_id = ? AND plant_date BETWEEN ? AND ?',
+        [projectId, startDate, `${endDate} 23:59:59.999`]
```

Comment:
Against which database you plan to run this SQL? No against the backend database I hope? I would strongly vote against directly trying to access it from the frontend instead of using API calls. Also I am not sure if the database access credentials would not be exposed in the JavaScript code. 

@jmiridis is this planned with you?

Reply:
@norbertschuler 
Database credentials are stored in environment variables, [which are then accessed by the package through the Node.js process.env object](https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/80380cf6088ab90a7678c905b948cbeb414cbc85/src/utils/connectDB.ts). 

The environment variables is set outside of the codebase, so it won't be exposed in JS code.

Reply:
I assumed if pushing some ENV value into `process.env` in `next.config.js` then it's available in the frontend code in the browser and this way getting public. At least I can print it to the console by adding `console.log("process.env.MYSQL_PASSWORD", process.env.MYSQL_PASSWORD)` e.g. into `pages/profile/treemapper/analytics`.tsx - if we do not push it ourselves into `process.env` then I would think it is private and only available for SSR? (Some env are even automatically made public - compare https://dumas-olivier.medium.com/keep-your-environment-variables-hidden-in-next-js-a06c143920f8.)